### PR TITLE
Support compilation with GHC 9

### PR DESCRIPTION
**Pull request by maksbotan**

### CODE REVIEW

File: src/Codec/Xlsx/Lens.hs [hunk 1]
```diff
@@ -84,7 +84,7 @@ ixCellRC i = wsCells . ix i
 -- | lens to access cell in a worksheet using more traditional
 -- x+y coordinates
 ixCellXY :: (Int, Int) -> Traversal' Worksheet Cell
-ixCellXY = ixCellRC . swap
+ixCellXY i = ixCellRC $ swap i
```
Comment (qrilka) [inline src/Codec/Xlsx/Lens.hs:87]: any hint why GHC 9 fails to compile it in point-free form?

File: src/Codec/Xlsx/Types/Drawing.hs [hunk 2]
```diff
@@ -217,6 +217,21 @@ data Anchor p g = Anchor
     } deriving (Eq, Show, Generic)
 instance (NFData p, NFData g) => NFData (Anchor p g)
 
+data GenericDrawing p g = Drawing
```
Comment (qrilka) [inline src/Codec/Xlsx/Types/Drawing.hs:220]: and why do we need this code block move?

File: src/Codec/Xlsx/Lens.hs (hunk 1 above)
Comment (maksbotan) [reply src/Codec/Xlsx/Lens.hs:87]: This is by design: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.0#simplified-subsumption
TLDR: to implement impredicativity they had to sacrifice some inference. In this case polymorhpic types do not unify anymore.

The error is:

```
src/Codec/Xlsx/Lens.hs:87:12: error:
    • Couldn't match type ‘c2’ with ‘Traversal' Worksheet Cell’
      Expected: (Int, Int) -> c2
        Actual: (Int, Int) -> Traversal' Worksheet Cell
      Cannot instantiate unification variable ‘c2’
      with a type involving polytypes: Traversal' Worksheet Cell
```

File: src/Codec/Xlsx/Types/Drawing.hs (hunk 2 above)
Comment (maksbotan) [reply src/Codec/Xlsx/Types/Drawing.hs:220]: Instances defined after TH splices are not seen before them: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.0#the-order-of-th-splices-is-more-important

In this case I had to move `makeLenses` after `Default` instance, but then this type started giving errors as well, so I had to move it before splice as well.

**Comment by qrilka** [comment]
I'm waiting for a new Stack release to land so I could add this into the project's CI. I hope it doesn't cause much of a trouble

**Comment by maksbotan** [comment]
No problem at all. I was only doing preliminary research, I won't consider switching to 9 until there is stackage nightly either.

**Comment by qrilka** [comment]
And with the nightly on GHC9 we're good to merge this, thanks @maksbotan !

**Comment by maksbotan** [comment]
Wonderful, thanks!

**Comment by qrilka** [comment]
Released as 0.8.4
